### DISCUSSION TITLE
Implement zero-RK in Julia

### DISCUSSION THREAD

**Discussion by jiweiqi** [LLNL/zero-rk#2] [Ideas]
**Abstract**

Just a casual thread here. The acceleration of zero-RK is pretty impressive. Recently, there are a couple of undergoing Julia packages for chemical kinetic modeling in Julia, motivated by the powerful auto-differentiation and adjoint sensitivity in Julia eco-system.

One of the pain we have been experiencing is the relatively slow adjoint computation due to the stiffness. I am wondering preconditioner in zero-RK might be useful for the adjoint method as well.

None of I and Matt knows preconditioner very well. If you are interested in Julia's implementations, we could start some discussions on how to accelerate the development.

**Reference**

https://github.com/ReactionMechanismGenerator/ReactionMechanismSimulator.jl/issues/71
https://github.com/ReactionMechanismGenerator/ReactionMechanismSimulator.jl
https://github.com/DENG-MIT/Arrhenius.jl

**Comment by whitesides1** [comment]
Having the ability for Zero-RK to perform faster adjoint sensitivity is one of the top, albeit unfunded, priorities for the next major update.  It would be helpful to have a conversation about your use case and other solver needs to work with Julia.  If you'd like, we can set up a conference call to discuss possible collaboration on this, just reach out to me and/or Matt at our LLNL e-mails:
({whitesides1,mcnenly1}@llnl.gov)
